pseudomonas_analysis/clustering_pa_data.py

```diff
@@ -0,0 +1,223 @@
+pa14_sample_ids_shared = set(raw_pa_compendium.index).intersection(pa14_sample_ids)
+
+# Label samples
+raw_pa_compendium["strain type"] = "other"
+raw_pa_compendium.loc[pao1_sample_ids_shared, "strain type"] = "PAO1"
+raw_pa_compendium.loc[pa14_sample_ids_shared, "strain type"] = "PA14"
+
+raw_pa_compendium.head()
+
+raw_pa_compendium_numeric = raw_pa_compendium.drop(columns=["strain type"])
+
+# ## Plot
+
+# +
+# UMAP embedding of original input data
+
+# pca = PCA(n_components=2)
+# model = pca.fit(raw_pa_compendium_numeric)
+
+# Get and save model
+model = umap.UMAP(random_state=random_state).fit(raw_pa_compendium_numeric)
+
+input_data_UMAPencoded = model.transform(raw_pa_compendium_numeric)
+input_data_UMAPencoded_df = pd.DataFrame(
+    data=input_data_UMAPencoded,
+    index=raw_pa_compendium_numeric.index,
+    columns=["1", "2"],
+)
+# Add label
+input_data_UMAPencoded_df["strain type"] = raw_pa_compendium["strain type"]
+
+input_data_UMAPencoded_df.head()
+# -
+
+# Colors
+marker_colors = {
+    "PA14": "#EF8B46",
+    "PAO1": "#C6A9B5",
+    "other": "#808080",
+}
+
+# +
+# Plot
+fig1 = ggplot(input_data_UMAPencoded_df, aes(x="1", y="2"))
+fig1 += geom_point(aes(color="strain type"), alpha=0.2, size=3, stroke=0.8)
+fig1 += scale_color_manual(values=marker_colors)
+fig1 += labs(x="UMAP 1", y="UMAP 2", title="Pa compendium in gene space")
+fig1 += theme_bw()
+fig1 += theme(
+    # figure_size=(6, 10),
+    legend_title_align="center",
+    plot_background=element_rect(fill="white"),
+    legend_key=element_rect(fill="white", colour="white"),
+    legend_title=element_text(family="sans-serif", size=15),
+    legend_text=element_text(family="sans-serif", size=12),
+    plot_title=element_text(family="sans-serif", size=15),
+    axis_text=element_text(family="sans-serif", size=12),
+    axis_title=element_text(family="sans-serif", size=15),
+)
+fig1 += guides(colour=guide_legend(override_aes={"alpha": 1}))
+
+print(fig1)
+
+fig1.save("pa_raw_clustering.svg", format="svg", dpi=300)
+
+# +
+# Load model
+vae_model_dir = "models/NN_2500_30/"
+
+# Files
+model_encoder_filename = glob.glob(os.path.join(vae_model_dir, "*_encoder_model.h5"))[0]
+
+weights_encoder_filename = glob.glob(
+    os.path.join(vae_model_dir, "*_encoder_weights.h5")
+)[0]
+
+model_decoder_filename = glob.glob(os.path.join(vae_model_dir, "*_decoder_model.h5"))[0]
+
+weights_decoder_filename = glob.glob(
+    os.path.join(vae_model_dir, "*_decoder_weights.h5")
+)[0]
+
+# Load saved models
+loaded_model = load_model(model_encoder_filename)
+loaded_decode_model = load_model(model_decoder_filename)
+
+loaded_model.load_weights(weights_encoder_filename)
+loaded_decode_model.load_weights(weights_decoder_filename)
+# -
+
+# Encode normalized expression data
+data_encoded = loaded_model.predict_on_batch(raw_pa_compendium_numeric)
+data_encoded_df = pd.DataFrame(data_encoded, index=raw_pa_compendium_numeric.index)
+
+# +
+# UMAP embedding of original input data
```

Comment:
Should this be "UMAP embedding of VAE encoded data", or something like that?

Reply:
Yes you're correct. I'll update this comment